packages/mobile/src/tokens/saga.ts

```diff
@@ -369,6 +369,34 @@ export function* watchFetchBalance() {
   yield takeEvery(fetchTokenBalances.type, fetchTokenBalancesSaga)
 }
 
+export function* watchAccountFundedOrLiquidated() {
+  let prevTokenBalance
+  while (true) {
+    const tokenBalance: ReturnType<typeof totalTokenBalanceSelector> = yield select(
+      totalTokenBalanceSelector
+    )
+    if (tokenBalance !== prevTokenBalance) {
+      // prevTokenBalance is undefined for the base case and null if token list
+      // is not yet loaded
+      if (prevTokenBalance) {
```

Comment:
i added this to prevent the account_funded event from being fired when a user is restoring their account (the balance goes from null to a non-zero amount in that case)